cuML/src/common/rmmAllocatorAdapter.hpp

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <rmm/rmm.h>
```

Comment:
So, we expect rmm now to be installed before cuml build, right?

Reply:
No. This header is just provided as a convenience for users of cuML. It should not be used inside of cuML. E.g. if a cuser of cuML wants to use RMM for all device allocations code like this could be used:
 ```cpp
#include <cuML.hpp>
#include <rmm/rmm.h>
#include <common/rmmAllocatorAdapter.hpp>

void foo() {
    rmmOptions_t rmmOptions;
    rmmOptions.allocation_mode = PoolAllocation;
    rmmOptions.initial_pool_size = 0;
    rmmOptions.enable_logging = false;
    rmmError_t rmmStatus = rmmInitialize(&rmmOptions);
    if ( RMM_SUCCESS != rmmStatus )
        std::cerr<<"WARN: Could not initialize RMM: "<<rmmGetErrorString(rmmStatus)<<std::endl;

    std::unique_ptr<ML::cumlHandle> cumlHandle( new ML::cumlHandle() );

    std::shared_ptr<ML::rmmAllocatorAdapter> allocator( new ML::rmmAllocatorAdapter() );

    cumlHandle->setDeviceAllocator( allocator );

    cudaStream_t stream;
    cudaStreamCreate( &stream );
    cumlHandle->setStream( stream );

    cudaStreamDestroy( stream );
    if ( rmmIsInitialized( NULL ) )
        rmmFinalize();
}
```

Reply:
@jirikraus thinking about it, could we put that `foo` function you just wrote (or something similar) somewhere as a convenience function for creating rmm enabled cumlHandle? From the python library I believe I will always call with RMM as the memory manager for the time being

Reply:
Ah! Makes sense now. Thanks Jiri.

Reply:
@dantegd maybe adding this as a code-block in the doxygen comment would be better for now?

Reply:
Yeah, probably will end up adding it for the cython call, but in doxygen is fine for now

Reply:
@teju85 I agree this should go into a code-block in the doxygen comment. However before we concluded the discussion on the API I don't think it makes sense to invest much in documentation, which is the reason I have not added documentation yet.
@dantegd I think this code should be only in the documentation and examples. It needs to be written by the user of cuML as that is the one who knows if RMM is already initialized and how he want's to initialize. E.g. if cuML gets called from a python script which used cuDF to prepare data RMM was initialized already before. If we exluce RMM init and shutdown the function would only provide 
```cpp
std::shared_ptr<ML::rmmAllocatorAdapter> allocator( new ML::rmmAllocatorAdapter() );
cumlHandle->setDeviceAllocator( allocator );
```
which IMHO is not worth a separate helper function. Do you agree?

Reply:
I agree with @jirikraus here. It is not worth such a helper function.

Sure Jiri. Makes sense to document this once we finalized the interfaces.